### DISCUSSION TITLE
Variable "data" does not exist inside => pimcore area block

### DISCUSSION THREAD

**Discussion by marrep** [pimcore/pimcore#9015] [Q&A / Support] [answered]
# bug description

I have created a **quiz block** which **works perfectly fine** as long as it is implemented inside the view.html.twig.
Next, I would like to put this long piece of code into a **separate pimcore area block**.
However, since I use a data-object inside the quiz-block I do receive the following alert...

`Variable "data" does not exist.
#0 /var/www/html/var/cache/dev/twig/e8/e8caf3295de9883bfd56e0f6c997f0cfc4836e3ad8ea8319c1aea78fb65c8c74.php(43): __TwigTemplate_1058968173ff3410cecd8ca524bc56847dbf0e60bda441e52854efee402c75f8->{closure}() #1 /var/www/html/vendor/twig/twig/src/Template.php(405): __TwigTemplate_1058968173ff3410cecd8ca524bc56847dbf0e60bda441e52854efee402c75f8->doDisplay(Array, Array) #2 /var/www/html/vendor/twig/twig/src/Template.php(378): Twig\Template->displayWithErrorHandling(Array, Array) #3 /var/www/html/vendor/twig/twig/src/Template.php(390): Twig\Template->display(Array) #4 /var/www/html/vendor/symfony/symfony/src/Symfony/Bridge/Twig/TwigEngine.php(54): Twig\Template->render(Array) #5`

### Code snippet from view.html.twig

`		{{ pimcore_area('myArea', 
				{ 
					'type' : 'quiz'
				}
			) 
		}}`

### Code snippet from Areas/quiz/view.html.twig

```
{% set quizzes = data %}
{% set quizname = document.getProperty('quizname') %}
{% for quiz in quizzes|filter(quiz => quiz.name == quizname) %}
<section class="quiz" style="background-color: {{ quiz.backgroundColor }}">
... 
</section>
{% endfor %}
```

### How do I have to refactor my code in order to outsource the quiz-block while passing it the quiz-object?

**Comment by dvesh3** [comment, marked as answer]
@marrep pass params from your view to brick https://pimcore.com/docs/pimcore/current/Development_Documentation/Documents/Editables/Areablock/index.html#page_Integrate-an-Areablock-in-a-Template
```twig
{{ pimcore_area('myArea', { 'type' : 'quiz', "params": { "data": data } ) }}
```

**Reply by marrep** [reply]
@dvesh3 I did already and tried to access the data by using
```
{% set quizzes = data %}
{% set quizzes = this.data %}
{% set quizzes = params.data %}
```
...all causing the same Variable issue " xxx does not exist"

**Reply by dvesh3** [reply]
sorry, you need to provide brick index in params like `{{ pimcore_area('myArea', { 'type' : 'quiz', "params": { "quiz": { "data" : data} }} ) }}` or use globalParams

**Reply by marrep** [reply]
thanks, this perfectly worked out!!
